digitaltwin_client/src/dt_interface_client.c

```diff
@@ -278,65 +287,184 @@ static bool DT_IsAlpha(char c)
             (c >= 'A' && c <= 'Z'));
 }
 
-// DT_InterfaceClient_CheckNameValid makes sure that the passed names (specifically on registration) is legal.
-// In general the SDK does NOT do policy checking like this, leaving it to the server.  We make an exception
-// in this case because an ill-formatted values on registration over MQTT will result in the
-// TCP connection being closed.  Because this can be so hard for developer to diagnose interface being
-// invalid versus network issues, we provide this check.
-int DT_InterfaceClient_CheckNameValid(const char* valueToCheck, bool isInterfaceId)
+// IsEndOfSegmentCharacter returns whether the current character is the end of a segment.
+// For component names, this will simply be a 0.  For interfaces, it will be a ';' or ':'.
+static bool IsEndOfSegmentCharacter(char c, DT_SEGMENT_TYPE segmentType)
+{
+    if (segmentType == DT_SEGMENT_TYPE_COMPONENT_NAME)
+    {
+        return (c==0);
+    }
+    else
+    {
+        return (c == DT_Segment_Separator || c == DT_Version_Separator);
+    }
+}
+
+// IsNameSegmentValid checks whether either a component name or a segment of
+// an interface is legal DTMI.  Legal requires the segment to start with an alphabetical character,
```

Comment:
Legal requires the segment  => Syntax requires the segment

Reply:
Wow, it's like Yoda wrote the original.  Fixing.